x-pack/plugins/security_solution/server/usage/endpoint/get_metrics.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { EndpointMetrics, UniqueEndpointCountResponse } from './types';
+import { ENDPOINT_METRICS_INDEX } from '../../../common/constants';
+import { tlog } from '../../lib/telemetry/helpers';
+
+export interface GetEndpointMetricsOptions {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+}
+
+export const getEndpointMetrics = async ({
+  esClient,
+  logger,
+}: GetEndpointMetricsOptions): Promise<EndpointMetrics> => {
+  return {
+    unique_endpoint_count: await getUniqueEndpointCount(esClient, logger),
+  };
+};
+
+export const getUniqueEndpointCount = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<number> => {
+  try {
+    const query: SearchRequest = {
+      expand_wildcards: ['open' as const, 'hidden' as const],
+      index: ENDPOINT_METRICS_INDEX,
```

Comment:
Looks good. I ran the following query on the Threat Intel monitoring cluster and it returned 8 endpoints.

```
GET .ds-metrics-endpoint.metrics-*/_search
{
  "size": 0,
  "query": {
    "range": {
      "@timestamp": {
        "gte": "now-24h",
        "lte": "now"
      }
    }
  },
  "aggs": {
    "endpoint_count": {
      "cardinality": {
        "field": "agent.id"
      }
    }
  }
}

==> 

{
  "took": 4,
  "timed_out": false,
  "_shards": {
    "total": 9,
    "successful": 9,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 8,
      "relation": "eq"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "endpoint_count": {
      "value": 8
    }
  }
}
```